pkg/controller/tas/topology_ungater.go

```diff
@@ -179,14 +180,44 @@ func (r *topologyUngater) Reconcile(ctx context.Context, req reconcile.Request)
 
 	psNameToTopologyRequest := workload.PodSetNameToTopologyRequest(wl)
 	allToUngate := make([]podWithUngateInfo, 0)
+	groupedPodSetAssignments := make(map[string][]*kueue.PodSetAssignment)
+
+	for i, psa := range wl.Status.Admission.PodSetAssignments {
+		groupName := strconv.Itoa(i)
+		if psNameToTopologyRequest[psa.Name] != nil && psNameToTopologyRequest[psa.Name].PodSetGroupName != nil {
+			groupName = *psNameToTopologyRequest[psa.Name].PodSetGroupName
+		}
+		groupedPodSetAssignments[groupName] = append(groupedPodSetAssignments[groupName], &psa)
+	}
+
+	rankOffsets := make(map[kueue.PodSetReference]int32)
+	maxRank := make(map[kueue.PodSetReference]int32)
+
+	for _, psas := range groupedPodSetAssignments {
+		if len(psas) > 1 {
+			leader := psas[0]
+			workers := psas[1]
+			if *leader.Count > *workers.Count {
+				leader = psas[1]
+				workers = psas[0]
+			}
```

Comment:
How do we know leader is always first? User can declare another other, right? I think the algo does not really care, so we can just say "largerPodSet" and "smallerPodSet", if it matters for the algo, does it?

Also, two podset is ok for LWS, but we may need to ticket this for supporting JobSet where more PodSets can have the same group name.